ClientApp/src/app/Shared.service.ts

```diff
@@ -0,0 +1,38 @@
+import { Injectable } from '@angular/core';
+import { BehaviorSubject } from 'rxjs';
+
+import { FlightForSearch } from './FlightForSearch';
+
+@Injectable({
+  providedIn: 'root',
+})
+
+export class SharedService {
+
+  empty1: FlightForSearch  = {
+    departure: 0,
+    arrival: 0,
+    date: new Date(),
+    amount: 1
+  };
+
+  empty2: FlightForSearch  = {
+    departure: 0,
+    arrival: 0,
+    date: new Date(),
+    amount: 1
+  };
```

Comment:
What do you need these empty objects for?

Reply:
They're needed to share info between components. They can't be null or undefined